src/common/runtime/src/runtime.rs

```diff
@@ -50,45 +46,42 @@ impl Drop for Dropper {
     }
 }
 
-impl Runtime {
-    pub fn builder() -> Builder {
+pub trait RuntimeTrait {
+    /// Get a runtime builder
+    fn builder() -> Builder {
```

Comment:
returning a fixed Builder may not make sense. In most case it should be

pub trait Runtime {
  type Builder;

  fn builder() -> Self::Builder;

}

Also consider removing Trait from trait name

Reply:
The builder is reused, only the fn build() need to be override

Reply:
Runtime is used by the type Runtime